src/NServiceBus.Callbacks/ConvertLegacyEnumResponseToLegacyControlMessageBehavior.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using NServiceBus.Features;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Pipeline.Contexts;
+
+    class ConvertLegacyEnumResponseToLegacyControlMessageBehavior : PhysicalOutgoingContextStageBehavior
+    {
+        public override void Invoke(Context context, Action next)
+        {
+            var instance = context.Get<object>("RequestResponse.OutgoingLogicalMessageInstance");
+
+            if (!CallbacksSupport.isLegacyEnumResponse(instance.GetType()))
```

Comment:
`context.MessageType` should be available on the context? (https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Pipeline/Contexts/PhysicalOutgoingContext.cs#L37)

It was added recently, rebase?